client/src/header.rs

```diff
@@ -0,0 +1,70 @@
+use crate::Route;
+use yew::prelude::*;
+use yew_router::prelude::*;
+
+#[function_component(Header)]
+pub fn header() -> Html {
+    let history = use_history().unwrap();
+
+    let home_button = {
+        let history = history.clone();
+        let onclick = Callback::once(move |_| history.push(Route::Home));
```

Comment:
this will possibly be used more often, maybe create function push_history(r: Route)?